### PR TITLE
Fix spurious warnings with CC and the `kotlin-dsl` plugin

### PR DESCRIPTION
**Pull request by eskatos**
By letting CC store task logger message rewriters

* Fixes https://github.com/gradle/gradle/issues/25483

### CODE REVIEW

**Comment by eskatos** [comment]
@bot-gradle test ACC APT

File: subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/DefaultContextAwareTaskLoggerCodec.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+
+object DefaultContextAwareTaskLoggerCodec : Codec<DefaultContextAwareTaskLogger> {
+
+    override suspend fun WriteContext.encode(value: DefaultContextAwareTaskLogger) {
+        write(value.messageRewriter)
+    }
+
+    override suspend fun ReadContext.decode(): DefaultContextAwareTaskLogger =
+        DefaultContextAwareTaskLogger(Logging.getLogger(Task::class.java)).also { logger ->
```
Comment (bamboo) [inline subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/DefaultContextAwareTaskLoggerCodec.kt:35]: 🤔 `decodePreservingIdentity { id -> ... }`

File: platforms/core-configuration/kotlin-dsl-plugins/src/integTest/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinCompilerWarningsTest.kt [hunk 2]
```diff
@@ -16,7 +16,6 @@
 
 package org.gradle.kotlin.dsl.plugins.dsl
 
-import org.gradle.integtests.fixtures.ToBeFixedForConfigurationCache
```
Comment (bamboo) [inline platforms/core-configuration/kotlin-dsl-plugins/src/integTest/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinCompilerWarningsTest.kt:19]: 🎉

**Comment by eskatos** [comment]
@bot-gradle test ACC APT

File: subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/beans/BeanSchema.kt [hunk 3]
```diff
@@ -188,7 +188,8 @@ val abstractTaskRelevantFields = listOf(
     "actions",
     "enabled",
     "timeout",
-    "onlyIfSpec"
+    "onlyIfSpec",
+    "logger",
```
Comment (bamboo) [inline subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/beans/BeanSchema.kt:192]: 👍

File: subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/DefaultContextAwareTaskLoggerCodec.kt [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import org.gradle.configurationcache.serialization.ReadContext
+import org.gradle.configurationcache.serialization.WriteContext
+import org.gradle.internal.logging.slf4j.ContextAwareTaskLogger.MessageRewriter
+import org.gradle.internal.logging.slf4j.DefaultContextAwareTaskLogger
+
+
+object DefaultContextAwareTaskLoggerCodec : Codec<DefaultContextAwareTaskLogger> {
+
+    override suspend fun WriteContext.encode(value: DefaultContextAwareTaskLogger) {
+        write(value.messageRewriter)
```
[the remark this answers is not in the document]
Comment (eskatos) [reply subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/DefaultContextAwareTaskLoggerCodec.kt]: Thanks for the pointers and refresh of my memory ;)
Done

File: subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/DefaultContextAwareTaskLoggerCodec.kt [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        }
+    }
+
+    override suspend fun ReadContext.decode(): DefaultContextAwareTaskLogger =
+        decodePreservingIdentity { id ->
+            DefaultContextAwareTaskLogger(Logging.getLogger(Task::class.java)).also { logger ->
+                read()?.let { rewriter ->
+                    logger.messageRewriter = rewriter as MessageRewriter
+                }
+                isolate.identities.putInstance(id, logger)
```
Comment (bamboo) [inline subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/DefaultContextAwareTaskLoggerCodec.kt:44]: 👍

**Review by bamboo — APPROVED** [review]

**Pull request merged**